File: PotenzaCoda.c
#include <stdio.h>

int potenza_helper(int a, int x, int parziale);
int potenza(int a, int x);

int main(void){

	int a, x;
	int pow;

	printf("Inserisci la potenza (es. 3^2): ");
	scanf("%d^%d", &a, &x);

	pow = potenza(a, x);
	printf("%d^%d = %d\n", a, x, pow);

	return 0;
}

int potenza(int a, int x){
	return potenza_helper(a, x, 1);
}

int potenza_helper(int a, int x, int parziale){

	if (x == 1) return parziale * a;

	return potenza_helper(a, x-1, parziale*a);
}

File: PotenzaCoda.py
def potenza(a, x):
	return potenza_helper(a, x, 1)

def potenza_helper(base, exp, parziale):

	if exp == 1:
		return parziale * base
	elif exp == 0:
		return 1

	return potenza_helper(base, exp-1, parziale*base)

# main
e = -1
while e < 0:
	a, x = input("Inserisci la potenza (es. 3^2): ").split('^')
	a = int(a)
	x = int(x)


power = potenza(a, x)

print(f"{a}^{x} = {power}\n")